xarray/backends/api.py

```diff
@@ -487,10 +489,40 @@ def maybe_decode_store(store, lock=False):
             )
             name_prefix = "open_dataset-%s" % token
             ds2 = ds.chunk(chunks, name_prefix=name_prefix, token=token)
-            ds2._file_obj = ds._file_obj
+
+        elif engine == "zarr":
+            # adapted from Dataset.Chunk() and taken from open_zarr
+            if not (isinstance(chunks, (int, dict)) or chunks is None):
+                if chunks != "auto":
+                    raise ValueError(
+                        "chunks must be an int, dict, 'auto', or None. "
+                        "Instead found %s. " % chunks
+                    )
+
+            if chunks == "auto":
+                try:
+                    import dask.array  # noqa
+                except ImportError:
+                    chunks = None
+
+            # auto chunking needs to be here and not in ZarrStore because
+            # the variable chunks does not survive decode_cf
+            # return trivial case
+            if not chunks:  # e.g. chunks is 0, None or {}
+                return ds
```

Comment:
open_zarr defaults to chunks="auto"
    open_dataset default to chunks=None

At one of our past meetings, the consensus was to keep `open_zarr` around with `chunks="auto"` as default for backward compatibility. 

`open_dataset(engine="zarr")` should use the current default `chunks` kwarg.

With that in mind can we avoid adding this stuff here?

Reply:
> ```
> open_zarr defaults to chunks="auto"
> open_dataset default to chunks=None
> ```

Happy with this :ok_hand:

> With that in mind can we avoid adding this stuff here?

Do you mean removing the `if chunks == "auto": ...` code block? I think it should be kept here in `open_dataset` (instead of in the `open_zarr` function) because users might want to do `open_mfdataset(engine="zarr", chunks-"auto")`, and so the `chunks="auto"` logic needs to be in `open_dataset`.